### PR TITLE
Mentor Profile Modal Fix

### LINKED ISSUE #221: Mentor Edit Profile Modal
**Describe the bug**
On the mentor profile page, when you open up the edit profile modal, there is a large space inside the modal where there's nothing, see screenshots below.

**Expected behavior**
Check the schedule appointment modal for the mentee side, that modal works flawlessly. I suspect this has to do with the different mechanisms used. The appointment modal uses forms while this buggy one does not(it uses Modal instead)

**Screenshots**
Screen Shot 2021-03-04 at 6 44 40 PM

### PR DESCRIPTION
**Pull request by Leundai**
## Description :star2:

Fixes #221 
I'm sorry about the large LOC count! :( Prettier formats all css files to two tabs

- Improved mobile compatibility of profile editing modal
- removed duplication of the profile page (page element was rendering twice)
- add usePersistentState hook for collapsed even after refresh
- moved useAuth to hooks folder
- fixed gallery css bugs and filter box height

## Screenshots
image

image

## TODOs :star:
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Review by AngelaLuo49021 — CHANGES_REQUESTED** [review]
Looks good but now the navbar on the side is always collapsed for me?

https://user-images.githubusercontent.com/42478474/112556685-b4a21a80-8d98-11eb-8ec6-23eb21a8c936.mov

File: frontend/src/components/MainNavHeader.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import { NavLink } from "react-router-dom";
+import { useMediaQuery } from "react-responsive";
+import { Layout } from "antd";
```
Comment (kenetec) [inline frontend/src/components/MainNavHeader.js:4]: You can delete this file

**Review by kenetec — APPROVED** [review]
Looks beautiful! Make sure to delete `MainNavHeader.js`, we won't need that

**Review by AngelaLuo49021 — APPROVED** [review]
Looks great now! <3

**Pull request merged**